doc/nrf/ug_thingy91.rst

```diff
@@ -0,0 +1,251 @@
+If the modem fails to connect using this preferred mode within the default timeout period (10 minutes), the modem switches to the other mode.
+
+Modem firmware
+==============
+
+The functionality of the multimode modem of Thingy:91 is based on its firmware which is available as a pre-compiled binary.
+You can download the modem firmware from the `Thingy:91 product website (downloads)`_.
+The downloadable zip file contains the full firmware image and the firmware image files for updating the firmware.
+
+.. _precompiled_fw:
+
+Programming pre-compiled firmware images
+****************************************
+
+Pre-compiled firmware image files are useful in the following scenarios:
+
+* Restoring the firmware to its initial image
+
+* Updating the application firmware to a newly released version
+
+* Updating the modem firmware
+
+Downloading pre-compiled firmware images
+========================================
+
+To obtain pre-compiled firmware images for firmware update, perform the following steps:
+
+	1. Go to the `Thingy:91 product description`_ page and under the **Downloads** tab, navigate to **Precompiled firmware**.
+	#. Download and extract the latest Thingy:91 firmware package.
+	#. Check the ``CONTENTS.txt`` file in the extracted folder for the location and names of the different firmware images.
+
+.. _programming_usb:
+
+Quick programming of pre-compiled firmware images
+=================================================
+
+When you have the pre-compiled firmware images ready, you can directly program the images onto the Thingy:91 using the `nRF Connect Programmer`_ app, which is implemented as part of the `nRF Connect for Desktop`_.
+
+This is a direct USB method, making use of the MCUboot feature and the inbuilt serial recovery mode of Thingy:91.
+You can program the Thingy:91 either on the nRF9160 SiP or on the nRF52840 SoC component.
+
+For the detailed procedure for programming a Thingy:91 using nRF Connect Programmer, see the following: (links to the thingy91 programming sections in nRF connect programmer userguide in infocenter to be added here).
+
+Updating modem firmware
+=======================
+
+You can update the modem firmware of Thingy:91 by using an external debug probe such as nRF9160 DK or any J-Link device supporting ARM Cortex-M33.
+See (link to the thingy91 modem update section in nRF connect programmer userguide in infocenter to be added here) for the detailed steps to update the modem firmware.
+
+.. _building_pgming:
+
+Building and programming from the source code
+*********************************************
+
+You can also program the Thingy:91 by using the images obtained by building the code in an |NCS| environment.
+
+To set up your system to be able to build a compatible firmware image, follow the `nRF Connect SDK Getting started guide`_.
+There are two build targets of interest for  Thingy:91 in |NCS|, namely:
+
+* ``nrf9160_pca20035ns``
```

Comment:
It's also possible to use the target `nrf9160_pca20035` for secure-only firmware. LTE/GPS can only be used with non-secure target.

Reply:
Placing this information in a table to include information about the secure and the non-secure versions. Also added the information ' LTE/GPS can only be used with non-secure target.' as a note.